### PR TITLE
Revert "podman build --pull: use correct policy"

### PR DESCRIPTION
**Pull request by vrothberg**
Also correct the help message and docs.

@TomSweeneyRedHat @rhatdan @baude PTAL

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

File: docs/source/markdown/podman-build.1.md [hunk 1]
```diff
@@ -455,9 +455,8 @@ not required for Buildah as it supports only Linux.
 
 #### **--pull**
 
-When the option is specified or set to "true", pull the image from the first
-registry it is found in as listed in registries.conf.  Raise an error if not
-found in the registries, even if the image is present locally.
+When the option is specified or set to "true", pull the image.  Raise an error
+if the image could not be pulled, even if the image is present locally.
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-build.1.md:459]: I prefer the old verbiage here due to the tie in with the registries.conf underpinnings.  But '<shrug>'.

Comment (vrothberg) [reply docs/source/markdown/podman-build.1.md:459]: I didn't like it as "registries" felt under specified, especially with the new aliases.

**Pull request merged**